services/management/dbaas/components_service.go

```diff
@@ -87,38 +96,113 @@ func (c componentsService) GetPXCComponents(ctx context.Context, req *dbaasv1bet
 			return nil, e
 		}
 
-		params.operatorVersion = checkResponse.Operators.Xtradb.Version
+		if checkResponse.Operators.Xtradb != nil {
+			params.operatorVersion = checkResponse.Operators.Xtradb.Version
+		}
 	}
 
-	versions, err := c.versions(ctx, params)
+	versions, err := c.versions(ctx, params, kubernetesCluster)
 	if err != nil {
 		return nil, err
 	}
 	return &dbaasv1beta1.GetPXCComponentsResponse{Versions: versions}, nil
 }
 
-func (c componentsService) versions(ctx context.Context, params componentsParams) ([]*dbaasv1beta1.Version, error) {
+func (c componentsService) ChangePSMDBComponents(ctx context.Context, req *dbaasv1beta1.ChangePSMDBComponentsRequest) (*dbaasv1beta1.ChangePSMDBComponentsResponse, error) {
+	err := c.db.InTransaction(func(tx *reform.TX) error {
```

Comment:
InTransactionContext

Reply:
I checked the code and we never used `InTransactionContext`, so if we will move to `InTransactionContext` we should do it everywhere for consistency

Reply:
It's ok to skip it for now then. Resolving the conversation.